src/main/java/com/ericsson/ei/config/ConfigurationLogger.java

```diff
@@ -1,78 +1,71 @@
-                + "email.sender: " + environment.getProperty("email.sender") + "\n"
-                + "email.subject: " + environment.getProperty("email.subject") + "\n"
-                + "notification.failAttempt: " + environment.getProperty("notification.failAttempt") + "\n"
-                + "notification.ttl.value: " + environment.getProperty("notification.ttl.value") + "\n"
-                + "spring.mail.host: " + environment.getProperty("spring.mail.host") + "\n"
-                + "spring.mail.port: " + environment.getProperty("spring.mail.port") + "\n"
-                + "spring.mail.username: " + environment.getProperty("spring.mail.username") + "\n"
-                + "spring.mail.properties.mail.smtp.auth: " + environment.getProperty("spring.mail.properties.mail.smtp.auth") + "\n"
-                + "spring.mail.properties.mail.smtp.starttls.enable: " + environment.getProperty("spring.mail.properties.mail.smtp.starttls.enable") + "\n"
-                + "er.url: " + environment.getProperty("er.url") + "\n"
-                + "ldap.enabled: " + environment.getProperty("ldap.enabled") + "\n"
-                + "ldap.server.list: " + environment.getProperty("ldap.server.list") + "\n"
-                + "logging.level.root: " + environment.getProperty("logging.level.root") + "\n"
-                + "logging.level.org.springframework.web: " + environment.getProperty("logging.level.org.springframework.web") + "\n"
-                + "logging.level.com.ericsson.ei: " + environment.getProperty("logging.level.com.ericsson.ei") + "\n"
-                + "\nThese properties are only some of the configurations, more configurations could have been provided.\n");
+        List<MapPropertySource> propertySources = findApplicationProperties();
+        for (MapPropertySource propertySource : propertySources) {
+            String propertiesLogMessage = createPropertiesLogMessage(propertySource);
+            LOGGER.info(propertiesLogMessage);
+        }
+    }
+
+    private List<MapPropertySource> findApplicationProperties() {
+        List<MapPropertySource> propertySources = new ArrayList<>();
+        for (PropertySource<?> propertySource : ((ConfigurableEnvironment) environment).getPropertySources()) {
+            if (propertySource.getName().contains("applicationConfig")) {
+                propertySources.add((MapPropertySource) propertySource);
+            }
+        }
+        return propertySources;
+    }
+
+    private String createPropertiesLogMessage(MapPropertySource propertySource) {
+        String[] propertyNames = propertySource.getPropertyNames();
+        Arrays.sort(propertyNames);
+        String propertiesLogMessage = String.format("\n##### %s #####\n", propertySource.getName());
+        for (String propertyName : propertyNames) {
+            String propertyValue = environment.getProperty(propertyName);
+            propertiesLogMessage += String.format("%s: %s\n", propertyName, propertyValue);
```

Comment:
Should we use a string builder here instead?

Reply:
I looked at StringBuilder now. Do you mean to use append bit by bit, like below?
logMessage.append("\n##### ");
logMessage.append(propertySource.getName());
logMessage.append(" #####\n");
This looks less elegant to me but perhaps you meant something different?

Reply:
No more like
```
logMessage.append(String.format("\n##### %s #####\n", propertySource.getName())); 

and 
logMessage.append(String.format("%s: %s\n", propertyName, propertyValue));
```

Reply:
Done